src/main/java/com/kelinci/mainapp/api/UserController.java

```diff
@@ -37,27 +31,38 @@ public String getListOfRegisteredUsers() {
 
     @DeleteMapping(value = "/registered/delete")
     public void deleteOurUser() {
-        registeredUser = null;
-        confirmedUser = null;
+        ourUser = null;
         listOfUsers.clear();
     }
 
     @PostMapping(value = "/registration")
     public void register(@RequestBody RegisterRequest request) {
 
-        Integer random = (int) (1000000 * Math.random());
-        String mailCode = Integer.toString(random);
-        System.out.println(mailCode);
-
-        final OurUser ourUser = new OurUser(request.getMail(), mailCode, false);
-
-        //wysylamy tutaj maila
-
-        registeredUser = ourUser;
-        listOfUsers.add(registeredUser);
-
+        final OurUser userFromRequest = new OurUser(request.getMail());
+        //in case the list is empty - add the userFromRequest with its randomly generated mailCode
+        if (listOfUsers.isEmpty()) {
+            String mailCode = generateMailCode();
+            userFromRequest.setMailCode(mailCode);
+            ourUser = userFromRequest;
+            listOfUsers.add(userFromRequest);
+            //in case the list is not empty, stream all email addresses from listOfUsers into a new List: listOfAllEmails
+            //the list maps and collects only email addresses from listOfUsers
+            //eventually if listOfAllEmails does not contain the email from the incoming user, generate mailCode and add user
+
+        } else {
+            List<String> listOfAllEmails = listOfUsers.stream().map(OurUser::getMail).collect(Collectors.toList());
+
+            if (!(listOfAllEmails.contains(userFromRequest.getMail()))) {
+
+                String mailCode = generateMailCode();
+                userFromRequest.setMailCode(mailCode);
+                ourUser = userFromRequest;
+                listOfUsers.add(userFromRequest);
+            }
+        }
     }
 
+
     @PostMapping(value = "/registered/confirm")
     public void confirm(@RequestBody ConfirmationRequest confirmation) {
 
```

Comment:
Trochę taki overKill z tworzeniem tego 'userToBeConfirmed'. Raczej go nie potrzebujesz (W sensie nowego obiektu typu OurUser. Po prostu gdy kod wejdzie Ci w if(areEmailsAndMailCodesTheSame(...)) to wtedy na tym elemencie, który wszedł do ifa, przekręcasz flagę isConfirmed, nie?

Reply:
racja - faktycznie wystarczy zmienić metodę areEmailsAndMailCodesTheSame() żeby brała confirmation. i userToBeChecked. (getMail oraz getMailCode) i je porównywała - zmieniłem w kodzie i faktycznie zostają 2 linijki. NIe ma potrzeby też wyrzucania i dodawania usera manualnie - po prostu flaga isConfirmed jest zmieniona bezpośrednio w liście. Sprytne.